src/connector/src/sink/doris_starrocks_connector.rs

```diff
@@ -151,11 +153,75 @@ impl HeaderBuilder {
         self
     }
 
+    /// Only used in Starrocks Transaction API
+    pub fn set_db(mut self, db: String) -> Self {
+        self.header.insert("db".to_string(), db);
+        self
+    }
+
+    /// Only used in Starrocks Transaction API
+    pub fn set_table(mut self, table: String) -> Self {
+        self.header.insert("table".to_string(), table);
+        self
+    }
+
     pub fn build(self) -> HashMap<String, String> {
         self.header
     }
 }
 
+/// Try getting BE url from a redirected response, returning `Ok(None)` indicates this request does
+/// not redirect.
+///
+/// The reason we handle the redirection manually is that if we let `reqwest` handle the redirection
+/// automatically, it will remove sensitive headers (such as Authorization) during the redirection,
+/// and there's no way to prevent this behavior.
+fn try_get_be_url(resp: &Response, fe_host: String) -> Result<Option<Url>> {
+    match resp.status() {
+        StatusCode::TEMPORARY_REDIRECT => {
+            let be_url = resp
+                .headers()
+                .get("location")
+                .ok_or_else(|| {
+                    SinkError::DorisStarrocksConnect(anyhow::anyhow!(
+                        "Can't get doris BE url in header",
+                    ))
+                })?
+                .to_str()
+                .context("Can't get doris BE url in header")
+                .map_err(SinkError::DorisStarrocksConnect)?
+                .to_string();
+
+            let mut parsed_be_url =
+                Url::parse(&be_url).map_err(|err| SinkError::DorisStarrocksConnect(err.into()))?;
+
+            if fe_host != LOCALHOST && fe_host != LOCALHOST_IP {
```

Comment:
Hmm, actually any IP of `127.` refers to localhost i.e. loop-back. Does SR guarantee to return `127.0.0.1`?

Reply:
Related PR https://github.com/risingwavelabs/risingwave/pull/16018